src/api/spec/models/workflow_spec.rb

```diff
@@ -244,6 +244,26 @@
       login user
     end
 
+    context 'with matching labels filter' do
+      let(:yaml) do
+        { steps: [{ branch_package: { source_project: project.name, source_package: package.name, target_project: project.name } }],
+          filters: { event: 'pull_request', labels: { only: ['duplicate'] } } }
+      end
+      let!(:workflow_run) { create(:workflow_run, :pull_request_labeled, token: token) }
+
+      it { expect { subject.call }.to change(WorkflowArtifactsPerStep, :count).by(1) }
+    end
+
+    context 'with unmatched labels filter' do
+      let(:yaml) do
+        { steps: [{ branch_package: { source_project: project.name, source_package: package.name, target_project: project.name } }],
+          filters: { event: 'pull_request', labels: { only: ['random-label'] } } }
+      end
+      let!(:workflow_run) { create(:workflow_run, :pull_request_labeled, token: token) }
+
+      it { expect { subject.call }.not_to change(WorkflowArtifactsPerStep, :count) }
+    end
+
```

Comment:
Apart from the 'with matching labels filter' and 'with unmatched labels filter' test cases, I would also add a test case for ignored labels.